Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleSceneView.swift

```diff
@@ -262,6 +262,31 @@ private extension WorldScaleSceneView {
     }
 }
 
+public extension WorldScaleSceneView {
+    /// Determines the scene point for the given screen point.
+    /// - Parameter screenPoint: The point in screen's coordinate space.
+    /// - Returns: The scene point corresponding to screen point.
+    private func arScreenToLocation(screenPoint: CGPoint) -> Point? {
+        // Use the `raycast` method to get the matrix of `screenPoint`.
+        guard let localOffsetMatrix = arViewProxy.raycast(from: screenPoint) else { return nil }
+        let originTransformationMatrix = cameraController.originCamera.transformationMatrix
+        let scenePointMatrix = originTransformationMatrix.adding(localOffsetMatrix)
+        // Create a camera from transformationMatrix and return its location.
+        return Camera(transformationMatrix: scenePointMatrix).location
+    }
+    
+    /// Sets the closure to call when the scene's nodes are updated.
+    func onARScreenSingleTapGesture(
+        perform action: @escaping (_ screenPoint: CGPoint, _ scenePoint: Point?) -> Void
+    ) -> Self {
+        self
+            .onSingleTapGesture { tapPoint in
+                let scenePoint = arScreenToLocation(screenPoint: tapPoint)
+                action(tapPoint, scenePoint)
+            } as! WorldScaleSceneView
```

Comment:
The table top view doesn't force cast it to the type itself and returned `some View`. Do you think it has value to force cast here, or returning an opaque type is fine?